net/Socket.cpp

```diff
@@ -1149,63 +1151,119 @@ bool ServerSocket::bind([[maybe_unused]] Type type, [[maybe_unused]] int port)
 #endif
 }
 
+bool ServerSocket::isRetryAcceptError(const int cause)
+{
+    switch(cause)
+    {
+        case EINTR:
+        case EAGAIN:        // == EWOULDBLOCK
+        case ENETDOWN:      // man accept(2): to be treated like EAGAIN
+        case EPROTO:        // man accept(2): to be treated like EAGAIN
+        case ENOPROTOOPT:   // man accept(2): to be treated like EAGAIN
+        case EHOSTDOWN:     // man accept(2): to be treated like EAGAIN
+#ifdef ENONET
+        case ENONET:        // man accept(2): to be treated like EAGAIN (undef on BSD, APPLE, ..)
+#endif
+        case EHOSTUNREACH:  // man accept(2): to be treated like EAGAIN
+        case EOPNOTSUPP:    // man accept(2): to be treated like EAGAIN
+        case ENETUNREACH:   // man accept(2): to be treated like EAGAIN
+        case ECONNABORTED:  // OK
+        case ETIMEDOUT:     // OK, should not happen due ppoll
+            return true;
+        default:
+            return false;
+    }
+}
+
 std::shared_ptr<Socket> ServerSocket::accept()
 {
     // Accept a connection (if any) and set it to non-blocking.
     // There still need the client's address to filter request from POST(call from REST) here.
 #if !MOBILEAPP
     assert(_type != Socket::Type::Unix);
 
+#if ENABLE_DEBUG
+    UnitWSD* const unitWsd = UnitWSD::isUnitTesting() ? &UnitWSD::get() : nullptr;
```

Comment:
Much better. But please git grep for UnitWSD::isUnitTesting - and notice the common pattern for whether we need to add ENABLE_DEBUG #ifdefs around - I would expect you to read four other call-sites before using a new method in your code - and then to produce something that fits in. Please remove the #if  block.

Reply:
I looked for `StreamSocket::simulateSocketError` as a boiler plate, 
used only `#if ENABLE_DEBUG` removing side-effects.
Yes, `UnitWSD::isUnitTesting()` is unconditionally build-in otherwise (your preference), changing it.
- I have looked at `WebSocketHandler` earlier, etc
- Personally I would have added a `constexpr` `isDebugEnabled()` expression to remove this specific fault injection from the product, while still covering code analysis by the compiler.

Note that the pattern `UnitWSD::get().isUnitTesting()` used is not ideal,
since both `get()` and `isUnitTesting()` are static methods of `UnitWSD`.